sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -195,6 +196,8 @@ abstract class BaseRelation {
    *  java.lang.String -> UTF8String
    *  java.lang.Decimal -> Decimal
    *
+   * If `needConversion` is `false`, buildScan() should return an [[RDD]] of [[InternalRow]]
```

Comment:
Yeah, I don't have a great suggestion either.  This just stuck out at the hackiest part to me.